### ISSUE TITLE
vips_colourspace: no known route between 'multiband' and 'srgb' error when resizing

### ISSUE THREAD

**Issue by mxshrestha** [jcupitt/libvips#525]
m using libvips 8.3.3, CentOS 6.4. I get the following error with the resize command for attached file.

```
vipsthumbnail ./tests/images/test-resize-libvips.jpg --size 400x400
vipsthumbnail: unable to thumbnail ./tests/images/test-resize-libvips.jpg
vips_colourspace: no known route between 'multiband' and 'srgb'
```

![test-resize-libvips](https://cloud.githubusercontent.com/assets/1475253/18528752/e3736182-7a8f-11e6-90cf-a7025279a4b1.jpg)

**Comment by jcupitt** [comment]
Strange, it's working for me with 8.3. Could you run `vipsheader` on your gif and paste the results? I see:

```
$ vipsheader -a unnamed.gif 
temp-0: 500x500 uchar, 4 bands, srgb, gifload
width: 500
height: 500
bands: 4
format: 0 - uchar
coding: 0 - none
interpretation: 22 - srgb
xoffset: 0
yoffset: 0
xres: 1.000000
yres: 1.000000
filename: "temp-0"
vips-loader: gifload
```

**Comment by mxshrestha** [comment]
This is what i get:
```
[cloud-user@ip-10-253-172-73 images]$ vipsheader -a test-resize-libvips.jpg 
test-resize-libvips.jpg: 500x500 uchar, 2 bands, rgb, magickload
width: 500
height: 500
bands: 2
format: 0 - uchar
coding: 0 - none
interpretation: 17 - rgb
xoffset: 0
yoffset: 0
xres: 1.000000
yres: 1.000000
filename: "test-resize-libvips.jpg"
vips-loader: magickload
magick-date:create: 2016-09-14T21:07:50+00:00
magick-date:modify: 2016-09-14T21:07:50+00:00
[cloud-user@ip-10-253-172-73 images]$ 
```

**Comment by jcupitt** [comment]
Oh dear, I'm sorry, this fell off my radar.

Your libvips has been built without gif support, so libvips has fallen back to using imagemagick to load the file. The IM loader has made a two-band image (grey plus alpha), but tagged it as RGB. `colourspace` is then struggling to convert that to RGB for the JPEG writer.

The simplest fix would be to enable gif support in your libvips. I guess you built it yourself? Check the output of configure carefully, and make sure you have `yes` next to gif. You could switch to vips 8.4 as well.

I'll see if I can make the magick loader work a little better.

**Comment by jcupitt** [comment]
Actually, this has been fixed in 8.4, the magick loader there is smarter about the types it tags images with. I'll close.

Thank you for reporting this @mxshrestha and I'm sorry it took me so long to respond.
